### ISSUE TITLE
Lints from any project are used in all projects in multi-project workspace

### ISSUE THREAD

**Issue by TheDutchMC** [rust-lang/rust-clippy#7342]
When working in a `workspace`, lints enabled in one subproject get applied in all projects, even if they themselves do not enable the lints.

## Reproduce
Workspace structure:
```
Root
+--- homelander/
+--- TestModule/
+--- Cargo.toml
```
Workspace `Cargo.toml`:
```toml
[workspace]
members = [
    "homelander",
    "TestModule"
]
```

Lint in `test_module`:
```rs
#![warn(clippy::cargo)]
```
No lints are specified in `test_module`.

When running `cargo clippy` from the root of the workspace, the bug occurs.

## I expected to see this happen
Clippy does not apply lints in one project to all projects in a workspace

## Instead, this happened:
Clippy applies lints enabled in one project to all projects
```
warning: package `test_module` is missing `package.description` metadata
  |
note: the lint level is defined here
 --> homelander/src/main.rs:7:9
  |
7 | #![warn(clippy::cargo)]
  |         ^^^^^^^^^^^^^
  = note: `#[warn(clippy::cargo_common_metadata)]` implied by `#[warn(clippy::cargo)]`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#cargo_common_metadata

warning: package `test_module` is missing `either package.license or package.license_file` metadata
```

### Meta

- `cargo clippy -V`: `clippy 0.1.52 (9bc8c42 2021-05-09)`
- `rustc -Vv`:
```
rustc 1.52.1 (9bc8c42bb 2021-05-09)
binary: rustc
commit-hash: 9bc8c42bb2f19e745a63f3445f1ac248fb015e53
commit-date: 2021-05-09
host: x86_64-unknown-linux-gnu
release: 1.52.1
LLVM version: 12.0.0
```

**Comment by TheDutchMC** [comment]
Note that `TestModule` depends via a path dependency on `homelander`. I suppose that is why it might string along the lints defined in `homelander` to the `TestModule` crate?
